velox/expression/tests/ExprTest.cpp

```diff
@@ -2237,3 +2236,60 @@ TEST_F(ExprTest, flatNoNullsFastPath) {
   ASSERT_FALSE(exprSet->exprs()[0]->supportsFlatNoNullsFastPath())
       << exprSet->toString();
 }
+
+TEST_F(ExprTest, commonSubExpressionWithEncodedInput) {
+  // This test case does a sanity check of the code path that re-uses
+  // precomputed results for common sub-expressions.
+  // TODO: consider adding metrics that expose use of optimizations like this.
```

Comment:
I wonder if existing stats are sufficient. My understand is that before the fix CSE would be evaluated on the first set of rows and results would be remembered; then it will be evaluates of the new set of rows, but results would not be remembered. Hence, before the fix the total number of rows evaluated would be higher then after fix.

https://facebookincubator.github.io/velox/develop/debugging/print-expr-with-stats.html

Reply:
Thank you for explaining. The TODO as written is a bit difficult to understand. Maybe remove the TODO and file a GItHub issue to explain in details what is missing.

Reply:
filed #2417 for this